### ISSUE TITLE
Improve error handling on data upload

### ISSUE THREAD

**Issue by yannisk2** [label-sleuth/label-sleuth#401]
**Is your feature request related to a problem?**
When uploading documents, if the user tries to upload a csv file that does not have the expected header row, the system throws  exceptions, which are shown both on the UI and on the console. As reading and interpreting those exceptions is hard, the system should ideally not throw an exception but instead catch the error and show on the UI an appropriate error message.

**What is the expected behavior?**
When the file to be uploaded does not contain in the first row a header named `text`, the system should not throw an exception, but instead show a message explaining that the header is missing (together with information on the required and optional column names). 

**Steps to reproduce the current behavior**
1. On Label Sleuth's workspace/dataset management screen, try to upload the following dataset, which does not include a `text` header: [data-without-headers.csv](https://github.com/label-sleuth/label-sleuth/files/10875456/data-without-headers.csv)
2. Label Sleuth currently will throw the following exceptions (paths shortened):
```
Traceback (most recent call last):
  File ".../lib/python3.8/site-packages/pandas/core/indexes/base.py", line 3802, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 165, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 5745, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 5753, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'text'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File ".../label_sleuth/app.py", line 182, in add_documents
    curr_app.orchestrator_api.add_documents_from_file(dataset_name, temp_file_path)
  File ".../label-sleuth/label_sleuth/orchestrator/orchestrator_api.py", line 978, in add_documents_from_file
    documents = CsvFileProcessor(dataset_name, temp_file_path).build_documents()
  File ".../label-sleuth/label_sleuth/data_access/processors/csv_processor.py", line 65, in build_documents
    df = df[df[self.text_col].apply(lambda x: len(x) > 0)]
  File ".../lib/python3.8/site-packages/pandas/core/frame.py", line 3807, in __getitem__
    indexer = self.columns.get_loc(key)
  File ".../lib/python3.8/site-packages/pandas/core/indexes/base.py", line 3804, in get_loc
    raise KeyError(key) from err
KeyError: 'text'
```

**Additional context (if any):**
We may need to check if a similar behavior exists in the uploading of labeled data, in which case we may need to implement this improvement there as well.

**Comment by williamtaggart97** [comment]
@yannisk2 @martinscooper is anyone currently working on this? I would like to give it a shot if its still up for grabs.

**Comment by martinscooper** [comment]
Hi @williamtaggart97! No one is working on this. You are welcomed (or any of your students) to take this. 

Some insights about error handling:
- The frontend expects an error object that has an error explanation on either an `error` or `title` field. I think the second is the prefered one. See [this file](https://github.com/label-sleuth/label-sleuth/blob/main/frontend/src/error/errorHandler.ts) to see how the frontend parses the error.
- On the backend side, when an endpoint fails, the error is commonly returned on an `except` block. As an example, the following is a valid error report:

```
return jsonify({"type": "category_id_error",
                        "title": f"category_id should be an integer (got {category_id}) "}), 400
```

Currently, if uploading documents fails the `traceback.format_exc()` is shown to the user, which is obviously not ideal. A more self-explainable and fine-grained error should be returned, like: `The uploaded csv has no headers`.

**Comment by williamtaggart97** [comment]
@alonh @arielge I'm taking a look at the error handling in the app.py file and it looks like title and error are being used interchangeably. Any thoughts -- it would be cleaner for the front-end if we only had one format for sending error messages

**Comment by alonh** [comment]
I agree. Let's use "title" everywhere.

**Comment by williamtaggart97** [comment]
Done. Check #455 for those updates.

**Comment by martinscooper** [comment]
I just merged #455. Thanks a lot @williamtaggart97. As part of this change, @williamtaggart97 not only, improved uploading labels error messages, but also standarized the format in which errors are reported.
